root/etc/s6-overlay/s6-rc.d/svc-tailscale-up/run

```diff
@@ -22,6 +22,16 @@ if [ -v TAILSCALE_BE_EXIT_NODE ]; then
     FLAGS="${FLAGS} --advertise-exit-node=${TS_BE_EXIT_NODE}"
 fi
 
+if [ -v TAILSCALE_USE_EXIT_NODE ]; then
+    echo "[!] using ${TAILSCALE_USE_EXIT_NODE} as an exit node."
+    FLAGS="${FLAGS} --exit-node=${TAILSCALE_USE_EXIT_NODE}"
+
+    if [ -v TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS ]; then
+        echo '[!] allowing exit node LAN access.'
+        FLAGS="${FLAGS} --exit-node-allow-lan-access=${TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS}"
+    fi
+fi
```

Comment:
Instead of checking that the env is set, we should also ensure its value is `true`

Something like:

```
if [ "${TAILSCALE_USE_EXIT_NODE}" = "true" ]; then
    echo "[!] using exit node."
    FLAGS="${FLAGS} --exit-node"

    if [ "${TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS}" = "true" ] || \
       [ "${TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS}" = "false" ]; then
        echo '[!] configuring exit node LAN access.'
        FLAGS="${FLAGS} --exit-node-allow-lan-access=${TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS}"
    else
        echo '[!] TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS is not set to true or false. Skipping this setting.'
    fi
else
    echo "[!] TAILSCALE_USE_EXIT_NODE is not set to true. Skipping exit node configuration."
fi
```

Reply:
In your example, wouldn't `[ "${TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS}" = "false" ]` never be true, given we are checking for true before executing that branch?

Reply:
Ah, good catch. Let's use `-n` check for non-empty.

Reply:
And change `TAILSCALE_USE_EXIT_NODE` to `TAILSCALE_EXIT_NODE`

Reply:
An empty value is valid and is used to disable the feature in the configuration. If we check for empty values we wouldn't be able to turn off the feature.

Reply:
I was under the impression that `up` required the complete set of desired settings, so omitting it should clear it out. If we were using `set` that would not be the case. But I tested this and checked `tailscale debug prefs` between and you're correct we need to set it to an empty value to disable it.

We could change it for both of the settings to provide the default value if they are unset, or maybe a better solution would be to supply `--reset` and only pass the desired settings.

Reply:
I feel like transparently passing whatever option the user configures is the best way versus trying to catch all the edge cases. Maybe there's room for another change to provide a `--reset` option, but I feel like that's out of scope for this specific PR. Thoughts?